src/app/ui/components/virtuoso.tsx

```diff
@@ -0,0 +1,45 @@
+import { ReactNode, useState } from 'react';
+
+import { Box } from 'leather-styles/jsx';
+
+import { useViewportMinWidth } from '@app/common/hooks/use-media-query';
+import { useOnResizeListener } from '@app/common/hooks/use-on-resize-listener';
+
+interface VirtuosoWrapper {
+  children: ReactNode;
+  hasFooter?: boolean;
+  isPopup?: boolean;
+}
+
+function vhToPixels(vh: string) {
+  const numericHeight = +vh.replace('vh', '');
+  return (numericHeight * window.innerHeight) / 100;
+}
+
+export function VirtuosoWrapper({ children, hasFooter, isPopup }: VirtuosoWrapper) {
+  const [key, setKey] = useState(0);
+  const isAtLeastMd = useViewportMinWidth('md');
+  const virtualHeight = isAtLeastMd ? '70vh' : '100vh';
+  const headerHeight = isPopup ? 230 : 80;
+  const footerHeight = hasFooter ? 95 : 0;
+  const heightOffset = headerHeight + footerHeight;
+  const height = vhToPixels(virtualHeight) - heightOffset;
+
+  const onResize = () => {
```

Comment:
I tried to be more precise and update to the exact height we wanted on re-size but it was buggy. 

Instead, here I force a re-render on re-size by changing the `key`. This means the wrapper height can adjust properly according to the parents `vh`